src/plugins/compression.cpp

```diff
@@ -0,0 +1,91 @@
+#include "compression.h"
+
+std::unique_ptr<Compression> Compression::create(std::string encoding) {
+#ifdef ENABLE_LZ4
+    if (encoding.compare("LZ4") == 0) {
+        return std::make_unique<Lz4Compression>();
+    }
+    else
+#endif /* ENABLE_LZ4 */
+    if (encoding.compare("None") == 0) {
+        return std::make_unique<NoCompression>();
+    }
+    else {
+        return std::unique_ptr<Compression>(nullptr);
```

Comment:
at least log a warning here.

Reply:
Didn't want to add the whole logger dependency just for this. I did however, add an appropriate remark in the documentation of the create function, that it defaults to nullptr in case of failure.

Reply:
is it useful not to end the simulation here? wouldn't this just create nullptr exceptions? or are you catching them somewhere?

Reply:
The Compression wrapper tries to remain independent of surrounding code (i.e. the solver). As such, the solver can check for the return value of the pointer, and react accordingly. Failing to acquire a wrapper object does not necessarily mean to abort the simulation; one could still try a different compression algorithm, or just skip compression entirely if instantiation goes wrong. This should be decided by the solver, not the C' wrapper, hence this solution.

One could also default to NoCompression class, which is always available.

Reply:
ok, I would actually suggest to default to the NoCompression class. Otherwise, some user will just use it and encounter segfaults.

Reply:
e.g. having a config file using "LZ4" but forgetting to compile with lz4 support will result in awkward errors. As a logger, you might simply use the logger of ls1?

Reply:
Ah ok gotcha. Best way to solve this is throwing an exception.
And when I start doing that, I should probably revise the other error handling too.
I'll try to fix all this and we can discuss it again.